src/nvim/api/window.c

```diff
@@ -439,39 +439,99 @@ Boolean nvim_win_is_valid(Window window)
 /// changed. The following restriction apply: `row`, `col` and `relative`
 /// must be reconfigured together. Only changing a subset of these is an error.
-void nvim_win_config(Window window, Integer width, Integer height,
-                     Dictionary options, Error *err)
+///
+/// @param      window  Window handle
+/// @param      options Dictionary of options
+/// @param[out] err     Error details, if any
+void nvim_win_set_config(Window window, Dictionary options, Error *err)
   FUNC_API_SINCE(6)
 {
   win_T *win = find_window_by_handle(window, err);
   if (!win) {
     return;
   }
   bool new_float = !win->w_floating;
-  width = width > 0 ? width: win->w_width;
-  height = height > 0 ? height : win->w_height;
   // reuse old values, if not overriden
   FloatConfig config = new_float ? FLOAT_CONFIG_INIT : win->w_float_config;
 
   if (!parse_float_config(options, &config, !new_float, err)) {
     return;
   }
+  config.height = config.height > 0 ? config.height : win->w_height;
+  config.width = config.width > 0 ? config.width : win->w_width;
   if (new_float) {
-    if (!win_new_float(win, (int)width, (int)height, config, err)) {
+    if (!win_new_float(win, config, err)) {
       return;
     }
     redraw_later(NOT_VALID);
   } else {
-    win_config_float(win, (int)width, (int)height, config);
+    win_config_float(win, config);
     win->w_pos_changed = true;
   }
 }
 
+/// Return window configuration.
+///
+/// Return a dictionary containing the same options that can be given to
+/// |nvim_open_win()|.
+///
+/// `relative` will be empty for normal windows.
+///
+/// @param window Window handle
+/// @param[out] err Error details, if any
+/// @return Window configuration
+Dictionary nvim_win_get_config(Window window, Error *err)
+  FUNC_API_SINCE(6)
+{
+  Dictionary rv = ARRAY_DICT_INIT;
+
+  win_T *wp = find_window_by_handle(window, err);
+  if (!wp) {
+    return rv;
+  }
+
+  PUT(rv, "width", INTEGER_OBJ(wp->w_float_config.width));
+  PUT(rv, "height", INTEGER_OBJ(wp->w_float_config.height));
+  PUT(rv, "focusable", BOOLEAN_OBJ(wp->w_float_config.focusable));
+  PUT(rv, "external", BOOLEAN_OBJ(wp->w_float_config.external));
+  PUT(rv, "anchor", STRING_OBJ(cstr_to_string(
+      float_anchor_str[wp->w_float_config.anchor])));
+
+  if (wp->w_float_config.relative == kFloatRelativeWindow) {
+    PUT(rv, "win", INTEGER_OBJ(wp->w_float_config.window));
+  }
+
+  if (wp->w_float_config.external) {
+    return rv;
+  }
+
+  PUT(rv, "row", FLOAT_OBJ(wp->w_float_config.row));
+  PUT(rv, "col", FLOAT_OBJ(wp->w_float_config.col));
+
+  if (wp->w_floating) {
+    switch (wp->w_float_config.relative) {
+      case kFloatRelativeEditor:
+        PUT(rv, "relative", STRING_OBJ(cstr_to_string("editor")));
+        break;
+      case kFloatRelativeWindow:
+        PUT(rv, "relative", STRING_OBJ(cstr_to_string("win")));
+        break;
+      case kFloatRelativeCursor:
+        PUT(rv, "relative", STRING_OBJ(cstr_to_string("cursor")));
```

Comment:
if there were a list similar to `float_anchor_str` then the `switch` could be eliminated:

    PUT(rv, "relative", STRING_OBJ(cstr_to_string(float_rel_str[wp->w_float_config.relative])));